lib/reservation_manager.rb

```diff
@@ -0,0 +1,67 @@
+require 'time'
+
+require_relative 'room'
+require_relative 'reservation'
+
+module Hotel
+  class ReservationManager
+    attr_reader :rooms, :reservations
+
+    def initialize(room_count)
+      @rooms = generate_rooms(room_count)
+      @reservations = []
+    end
+
+    # Parameters: 
+    # -type: symbol (:SINGLE, :BLOCK) that indicates type of reservation
+    # -range: DateRange object 
+    # -occupancy: Hash or String
+    # Returns: newly created Reservation object
+    def create_reservation(type, range, occupancy)
+      raise ArgumentError.new("Invalid reservation type (given #{type}, must be :SINGLE or :BLOCK)") unless 
+        type == :SINGLE || type == :BLOCK
+
+      occupancy = [{:room => @rooms.sample, :guest => occupancy}] if occupancy.class == String
+
+      new_reservation = Reservation.new(type, range, occupancy)
+      @reservations << new_reservation
+      return new_reservation
+    end
+
+    # Parameters: Date object 
+    # Returns: an Array of all Reservation instances that contain that date
+    def find_reservations_by_date(check_in, check_out)
+      by_date = @reservations.select { |reservation| reservation.dates.overlap?(check_in, check_out)}
```

Comment:
This works because of implicit return, but I don't think you meant to assign here (since you don't use the variable):

```suggestion
      return @reservations.select { |reservation| reservation.dates.overlap?(check_in, check_out)}
```